src/meta/api/src/background_api_impl.rs

```diff
@@ -0,0 +1,368 @@
+                        BackgroundJobAlreadyExists::new(
+                            &name_key.name,
+                            format!("create background job: {:?}", req.job_name),
+                        ),
+                    )))
+                };
+            }
+
+            let id = fetch_id(self, IdGenerator::background_job_id()).await?;
+            let id_key = BackgroundJobId { id };
+
+            debug!(
+                id = debug(&id_key),
+                name_key = debug(&name_key),
+                "new backgroundjob id"
+            );
+
+            {
+                let meta: BackgroundJobInfo = req.job_info.clone();
+                let condition = vec![txn_cond_seq(name_key, Eq, 0)];
+                let if_then = vec![
+                    txn_op_put(name_key, serialize_u64(id)?), // name -> background_job_id
+                    txn_op_put(&id_key, serialize_struct(&meta)?), // id -> meta
+                ];
+
+                let txn_req = TxnRequest {
+                    condition,
+                    if_then,
+                    else_then: vec![],
+                };
+
+                let (succ, _responses) = send_txn(self, txn_req).await?;
+
+                debug!(
+                    name = debug(&name_key),
+                    id = debug(&id_key),
+                    succ = display(succ),
+                    "create_background_job"
+                );
+
+                if succ {
+                    break id;
+                }
+            }
+        };
+
+        Ok(CreateBackgroundJobReply { id })
+    }
+
+    // TODO(zhihanz): needs to drop both background job and related background tasks, also needs to gracefully shutdown running queries
+    async fn drop_background_job(
+        &self,
+        _req: DeleteBackgroundJobReq,
+    ) -> Result<DeleteBackgroundJobReply, KVAppError> {
+        todo!()
+    }
+
+    async fn update_background_job(
+        &self,
+        req: UpdateBackgroundJobReq,
+    ) -> Result<UpdateBackgroundJobReply, KVAppError> {
+        debug!(req = debug(&req), "BackgroundApi: {}", func_name!());
+        let name_key = &req.job_name;
+        let ctx = &func_name!();
+        let mut trials = txn_trials(None, ctx);
+
+        loop {
+            trials.next().unwrap()?;
+            let (seq, id) = get_u64_value(self, name_key).await?;
+            debug!(seq, id, ?name_key, "update_background_job");
+            let id_key = match seq {
+                0 => {
+                    let id = fetch_id(self, IdGenerator::background_job_id()).await?;
+                    BackgroundJobId { id }
+                }
+                _ => BackgroundJobId { id },
+            };
+            let meta = req.info.clone();
+            let condition = vec![txn_cond_seq(name_key, Eq, seq)];
+            let if_then = vec![
+                txn_op_put(name_key, serialize_u64(id_key.id)?),
+                txn_op_put(&id_key, serialize_struct(&meta)?),
+            ];
```

Comment:
For a indirect mapping `name -> id -> value`, an update operation handles the `id -> value` part, thus it's unnecessary to assert the seq of `name`(`txn_cond_seq(name_key, Eq, seq)`), but instead, it should assert that the seq of `id -> value` does not change during running this transaction.